gupnp-tools : Build Failiure

### Nixpkgs version

- Unstable (26.05)

### Describe the bug

https://hydra.nixos.org/build/314975066

```
Running phase: unpackPhase
unpacking source archive /nix/store/l6cxhci6wzan0lrhrnnv7pah43sjrcq8-gupnp-tools-0.12.2.tar.xz
source root is gupnp-tools-0.12.2
setting SOURCE_DATE_EPOCH to timestamp 1741718277 of file "gupnp-tools-0.12.2/subprojects/gupnp-av-1.0.wrap"
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
mesonConfigurePhase flags: --prefix=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2 --libdir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/lib --libexecdir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/libexec --bindir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/bin --sbindir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/sbin --includedir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/include --mandir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/share/man --infodir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/share/info --localedir=/nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/share/locale -Dauto_features=enabled -Dwrap_mode=nodownload --buildtype=plain
The Meson build system
Version: 1.9.1
Source dir: /private/tmp/nix-build-gupnp-tools-0.12.2.drv-0/gupnp-tools-0.12.2
Build dir: /private/tmp/nix-build-gupnp-tools-0.12.2.drv-0/gupnp-tools-0.12.2/build
Build type: native build
meson_options.txt:1: WARNING: Project does not target a minimum version but uses feature deprecated since '1.1.0': "boolean option" keyword argument "value" of type str. use a boolean, not a string
meson_options.txt:2: WARNING: Project does not target a minimum version but uses feature deprecated since '1.1.0': "boolean option" keyword argument "value" of type str. use a boolean, not a string
Project name: gupnp-tools
Project version: 0.12.2
C compiler for the host machine: clang (clang 21.1.2 "clang version 21.1.2")
C linker for the host machine: clang ld64 954.16
Host machine cpu family: aarch64
Host machine cpu: aarch64
Found pkg-config: YES (/nix/store/hygaaqwk9ylklp0ybwppqhw75nz8ya41-pkg-config-wrapper-0.29.2/bin/pkg-config) 0.29.2
Run-time dependency gssdp-1.6 found: YES 1.6.4
Run-time dependency gupnp-1.6 found: YES 1.6.9
Run-time dependency libsoup-3.0 found: YES 3.6.5
Run-time dependency gupnp-av-1.0 found: YES 0.14.4
Run-time dependency gtk+-3.0 found: YES 3.24.51
Run-time dependency glib-2.0 found: YES 2.86.1
Run-time dependency gio-2.0 found: YES 2.86.1
Run-time dependency libxml-2.0 found: YES 2.15.1
Run-time dependency gobject-2.0 found: YES 2.86.1
Run-time dependency gtksourceview-4 found: YES 4.8.4
meson.build:31: WARNING: Project does not target a minimum version but uses feature deprecated since '0.56.0': meson.source_root. use meson.project_source_root() or meson.global_source_root() instead.
Compiler for C supports arguments -Werror=deprecated-declarations: YES 
Configuring config.h using configuration
Configuring didl-lite-dialog.ui using configuration
Program msgfmt found: YES (/nix/store/69pxcqvnlmdwmd8v2x0nld0iw354nc11-gettext-0.25.1/bin/msgfmt)
Dependency gio-2.0 found: YES 2.86.1 (cached)
Program /nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources found: YES (/nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources)
Dependency gio-2.0 found: YES 2.86.1 (cached)
Program /nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources found: YES (/nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources)
src/av-cp/meson.build:12: WARNING: Project does not target a minimum version but uses feature deprecated since '0.56.0': gui_app arg in executable. Use 'win_subsystem' instead
Dependency gio-2.0 found: YES 2.86.1 (cached)
Program /nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources found: YES (/nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources)
src/network-light/meson.build:11: WARNING: Project does not target a minimum version but uses feature deprecated since '0.56.0': gui_app arg in executable. Use 'win_subsystem' instead
Dependency gio-2.0 found: YES 2.86.1 (cached)
Program /nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources found: YES (/nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/bin/glib-compile-resources)
src/universal-cp/meson.build:11: WARNING: Project does not target a minimum version but uses feature deprecated since '0.56.0': gui_app arg in executable. Use 'win_subsystem' instead
Program msginit found: YES (/nix/store/69pxcqvnlmdwmd8v2x0nld0iw354nc11-gettext-0.25.1/bin/msginit)
Program msgmerge found: YES (/nix/store/69pxcqvnlmdwmd8v2x0nld0iw354nc11-gettext-0.25.1/bin/msgmerge)
Program xgettext found: YES (/nix/store/69pxcqvnlmdwmd8v2x0nld0iw354nc11-gettext-0.25.1/bin/xgettext)
Build targets in project: 46
WARNING: Deprecated features used:
 * 0.56.0: {'gui_app arg in executable', 'meson.source_root'}
 * 1.1.0: {'"boolean option" keyword argument "value" of type str'}

gupnp-tools 0.12.2

  User defined options
    auto_features: enabled
    bindir       : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/bin
    buildtype    : plain
    includedir   : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/include
    infodir      : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/share/info
    libdir       : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/lib
    libexecdir   : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/libexec
    localedir    : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/share/locale
    mandir       : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/share/man
    prefix       : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2
    sbindir      : /nix/store/dafnq2gja43vcdni8839rfhryg28qmfv-gupnp-tools-0.12.2/sbin
    wrap_mode    : nodownload

Found ninja-1.13.1 at /nix/store/6y79vrqachzns4pgl8r0j04b7g33hl52-ninja-1.13.1/bin/ninja
mesonConfigurePhase: enabled\ parallel\ building
Running phase: buildPhase
build flags: -j2
[1/74] Generating src/common/common-resource_h with a custom command
[2/74] Generating src/common/common-resource_c with a custom command
[3/74] Compiling C object src/common/libutil.a.p/meson-generated_.._common-resource.c.o
[4/74] Compiling C object src/common/libutil.a.p/pretty-print.c.o
FAILED: [code=1] src/common/libutil.a.p/pretty-print.c.o 
clang -Isrc/common/libutil.a.p -Isrc/common -I../src/common -I. -I.. -I/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2 -I/nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/include -I/nix/store/9l57268nckpacngwqz6ahadwzax22ja1-glib-2.86.1-dev/include/glib-2.0 -I/nix/store/fhxjbdk0wmihyz8fjwmspswh6v9nkbhk-glib-2.86.1/lib/glib-2.0/include -I/nix/store/33s6y1fik8p4svscm6ki8hr3knvi766k-libsoup-3.6.5-dev/include/libsoup-3.0 -I/nix/store/hp6zqs6frcqsy7r8hf0x50slxb5dhjvv-gupnp-1.6.9-dev/include/gupnp-1.6 -I/nix/store/0hc38dn31jyny12psdqypq1ql9hw5hzx-gssdp-1.6.4-dev/include/gssdp-1.6 -I/nix/store/r0ri4x7n9kp87ls49x4ssc06s5xv6mv2-gtk+3-3.24.51-dev/include/gtk-3.0 -I/nix/store/sk8mal3v19sh5nh2g0yh5r2qwqa7c5qp-at-spi2-core-2.58.1-dev/include/atk-1.0 -I/nix/store/41p60wf0lz599g075ipl97zs5y8lf1pm-cairo-1.18.4-dev/include/cairo -I/nix/store/qi6gnavjkgnq06h70vlvbd4iy1kznr56-freetype-2.13.3-dev/include/freetype2 -I/nix/store/qi6gnavjkgnq06h70vlvbd4iy1kznr56-freetype-2.13.3-dev/include -I/nix/store/vln5snglwa320cff1ph82zaws0jc63dz-gdk-pixbuf-2.44.4-dev/include/gdk-pixbuf-2.0 -I/nix/store/l7x12g2mp48w7dm1x5rbbs9n0m7djbl8-pango-1.57.0-dev/include/pango-1.0 -I/nix/store/7dxm7gv35qxqqzxfkk42gf0kxblm841q-harfbuzz-12.1.0-dev/include/harfbuzz -fdiagnostics-color=always -Wall -Winvalid-pch -Werror=deprecated-declarations -MD -MQ src/common/libutil.a.p/pretty-print.c.o -MF src/common/libutil.a.p/pretty-print.c.o.d -o src/common/libutil.a.p/pretty-print.c.o -c ../src/common/pretty-print.c
../src/common/pretty-print.c:47:21: error: '__xmlIndentTreeOutput' is deprecated [-Werror,-Wdeprecated-declarations]
   47 |         old_value = xmlIndentTreeOutput;
      |                     ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/parser.h:1356:35: note: expanded from macro 'xmlIndentTreeOutput'
 1356 |     #define xmlIndentTreeOutput (*__xmlIndentTreeOutput())
      |                                   ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/parser.h:1285:1: note: '__xmlIndentTreeOutput' has been explicitly marked deprecated here
 1285 | XML_DEPRECATED
      | ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/xmlexports.h:63:43: note: expanded from macro 'XML_DEPRECATED'
   63 |     #define XML_DEPRECATED __attribute__((deprecated))
      |                                           ^
../src/common/pretty-print.c:48:9: error: '__xmlIndentTreeOutput' is deprecated [-Werror,-Wdeprecated-declarations]
   48 |         xmlIndentTreeOutput = 1;
      |         ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/parser.h:1356:35: note: expanded from macro 'xmlIndentTreeOutput'
 1356 |     #define xmlIndentTreeOutput (*__xmlIndentTreeOutput())
      |                                   ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/parser.h:1285:1: note: '__xmlIndentTreeOutput' has been explicitly marked deprecated here
 1285 | XML_DEPRECATED
      | ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/xmlexports.h:63:43: note: expanded from macro 'XML_DEPRECATED'
   63 |     #define XML_DEPRECATED __attribute__((deprecated))
      |                                           ^
../src/common/pretty-print.c:54:9: error: '__xmlIndentTreeOutput' is deprecated [-Werror,-Wdeprecated-declarations]
   54 |         xmlIndentTreeOutput = old_value;
      |         ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/parser.h:1356:35: note: expanded from macro 'xmlIndentTreeOutput'
 1356 |     #define xmlIndentTreeOutput (*__xmlIndentTreeOutput())
      |                                   ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/parser.h:1285:1: note: '__xmlIndentTreeOutput' has been explicitly marked deprecated here
 1285 | XML_DEPRECATED
      | ^
/nix/store/5gqh047ripng0qjbzkqv42q1p49caj2v-libxml2-2.15.1-dev/include/libxml2/libxml/xmlexports.h:63:43: note: expanded from macro 'XML_DEPRECATED'
   63 |     #define XML_DEPRECATED __attribute__((deprecated))
      |                                           ^
3 errors generated.
[5/74] Compiling C object src/common/libutil.a.p/icons.c.o
ninja: build stopped: subcommand failed.

```

### Steps to reproduce

curl https://hydra.nixos.org/build/314975066/reproduce | bash 

### Expected behaviour

Build was supposed to pass

### Screenshots

_No response_

### Relevant log output

```console

```

### Additional context

_No response_

### System metadata

➜  ~ nix-shell -p nix-info --run "nix-info -m"
 - system: `"aarch64-darwin"`
 - host os: `Darwin 25.0.0, macOS 26.0.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.31.2`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/store/hhyjq4dr4w4dwssm8vzq4cfjzxixhanp-source`

### Are you using nix-darwin?

Yes, I am using nix-darwin.

### Notify maintainers

_No response_

### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+darwin%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gu/gupnp-tools/package.nix
+++ b/pkgs/by-name/gu/gupnp-tools/package.nix
@@ -42,6 +42,11 @@ stdenv.mkDerivation (finalAttrs: {
     gtksourceview4
   ];
 
+  mesonFlags = [
+    # Work around https://gitlab.gnome.org/GNOME/gupnp-tools/-/issues/29
+    "-Dc_args=-Wno-error=deprecated-declarations"
+  ];
+
   passthru = {
     updateScript = gnome.updateScript {
       packageName = "gupnp-tools";

